run_unittests.py

```diff
@@ -4472,19 +4472,43 @@ def test_compiler_check_flags_order(self):
 
     def _test_stds_impl(self, testdir, compiler, p):
         lang_std = p + '_std'
-        # Check that all the listed -std=xxx options for this compiler work
-        # just fine when used
+        clangLT5 = (compiler.get_id() == 'clang' and
```

Comment:
I think I would simplify this by doing:
```python
clangLT5 = False
clangLT6 = False
clangLT8 = False
if compiler.get_id() == 'clang':
    clangLT8 = ...
    if clangLT8:
        clangLT6 = ...
        if clangLT5 = ...
```

Since we can infer that if clang is > 8 it's also greater than 6 and 5.

Reply:
Alternatively you might consider doing something like:
```python
# TODO: we don't have specialized rules for compilers other than gcc and clang.
has_cpp17 = compiler.get_Id() not in {'clang', 'gcc'} or ((compiler.get() == 'clang' and ...) or compiler.get_id() == 'gcc' and ...))
...

for v in ...:
    if '++17' in v:
        if not has_cpp17:
            continue
```

Reply:
Yes the "has_" logic is much clearer to read, great!